incubator/hnc/pkg/controllers/hierarchy_controller.go

```diff
@@ -262,10 +263,27 @@ func (r *HierarchyReconciler) syncLabel(log logr.Logger, nsInst *corev1.Namespac
 		return
 	}
 
+	// pre-define HNC label suffix
+	labelSuffix := fmt.Sprintf(".tree.%s/depth", api.MetaGroup)
+
+	// Remove all existing HNC labels.
+	for k := range nsInst.Labels {
+		if strings.HasSuffix(k, labelSuffix) {
+			delete(nsInst.Labels, k)
+		}
+	}
+
+	// Remove all existing HNC annotations.
+	for k := range nsInst.Annotations {
+		if strings.HasSuffix(k, labelSuffix) {
+			delete(nsInst.Annotations, k)
+		}
+	}
+
 	// AncestoryNames includes the namespace itself.
 	ancestors := ns.AncestoryNames(nil)
 	for i, ancestor := range ancestors {
-		labelDepthSuffix := ancestor + ".tree." + api.MetaGroup + "/depth"
+		labelDepthSuffix := ancestor + labelSuffix
```

Comment:
We should fix the variable name (I missed it last time) - this isn't the suffix, it's the entire label. So instead of `labelDepthSuffix`, let's change it to `depthLabel` or even just `label` or `l`, to be honest (see [here](https://github.com/golang/go/wiki/CodeReviewComments#variable-names)).

Reply:
Nice suggestion, i have update it